.github/workflows/unit-tests.yml

```diff
@@ -0,0 +1,17 @@
+name: Project Unit Tests
+
+on: [ push ]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Set up JDK 21
+        uses: actions/setup-java@v4
+        with:
+          java-version: '21'
+          distribution: 'adopt'
+      - uses: actions/checkout@v4
+      - name: Unit Tests with Gradle
+        run: ./gradlew clean test -Pgithub.packages.read.username=${{ secrets.GH_TG_LIB_USERNAME }} -Pgithub.packages.read.token=${{ secrets.GH_TG_LIB_TOKEN }}
```

Comment:
you don't need these params `-Pgithub.packages.read.username=${{ secrets.GH_TG_LIB_USERNAME }} -Pgithub.packages.read.token=${{ secrets.GH_TG_LIB_TOKEN }}`